apps/api/v2/src/modules/workflows/inputs/create-workflow.input.ts

```diff
@@ -0,0 +1,229 @@
+  EMAIL_HOST = "EMAIL_HOST",
+  EMAIL_ATTENDEE = "EMAIL_ATTENDEE",
+  EMAIL_ADDRESS = "EMAIL_ADDRESS",
+  SMS_ATTENDEE = "SMS_ATTENDEE",
+  SMS_NUMBER = "SMS_NUMBER",
+  WHATSAPP_ATTENDEE = "WHATSAPP_ATTENDEE",
+  WHATSAPP_NUMBER = "WHATSAPP_NUMBER",
+}
+
+export enum TemplateType {
+  REMINDER = "REMINDER",
+  CUSTOM = "CUSTOM",
+}
+
+export class WorkflowActivationDto {
+  @ApiProperty({
+    description: "Whether the workflow is active for all calendars",
+    example: false,
+    type: Boolean,
+  })
+  @IsBoolean()
+  isActiveOnAll!: boolean;
+
+  @ApiPropertyOptional({
+    description: "List of active calendar IDs the workflow applies to, required if isActiveOnAll is false",
+    example: [698191],
+    type: [Number],
+  })
+  @ValidateIf((o) => !Boolean(o.isActiveOnAll))
+  @IsOptional()
+  @IsNumber({}, { each: true })
+  activeOnEventTypeIds?: number[] = [];
+}
+
+export class WorkflowTriggerOffsetDto {
+  @ApiProperty({ description: "Time value for offset before/after event trigger", example: 24, type: Number })
+  @IsNumber()
+  value!: number;
+
+  @ApiProperty({ description: "Unit for the offset time", example: "HOUR", enum: WorkflowTimeUnit })
+  @IsEnum(WorkflowTimeUnit)
+  unit!: WorkflowTimeUnit;
+}
+
+export class WorkflowTriggerDto {
+  @ApiProperty({
+    description: "Trigger type for the workflow",
+    example: "BEFORE_EVENT",
+    enum: WorkflowTriggerType,
+  })
+  @IsEnum(WorkflowTriggerType)
+  type!: WorkflowTriggerType;
+
+  @ApiPropertyOptional({
+    description: "Offset before/after the trigger time; required for BEFORE_EVENT and AFTER_EVENT only",
+    type: WorkflowTriggerOffsetDto,
+    required: false,
+  })
+  @ValidateIf((o) =>
+    Boolean(
+      o.type && (o.type === WorkflowTriggerType.BEFORE_EVENT || o.type === WorkflowTriggerType.AFTER_EVENT)
+    )
+  )
+  @ValidateNested()
+  @Type(() => WorkflowTriggerOffsetDto)
+  @IsOptional()
+  offset?: WorkflowTriggerOffsetDto;
+}
+
+export class WorkflowMessageDto {
+  @ApiProperty({
+    description: "Subject of the message",
+    example: "Reminder: Your Meeting {EVENT_NAME} - {EVENT_DATE_ddd, MMM D, YYYY h:mma} with Cal.com",
+  })
+  @IsString()
+  subject!: string;
+
+  @ApiProperty({
+    description: "HTML content of the message",
+    example:
+      "<p>This is a reminder from {ORGANIZER} of {EVENT_NAME} to {ATTENDEE} starting here  {LOCATION} {MEETING_URL} at {START_TIME_h:mma} {TIMEZONE}.</p>",
+    required: false,
+  })
+  @IsOptional()
+  @IsString()
+  html?: string;
+
+  @ApiProperty({
+    description: "Text content of the message (used for SMS)",
+    example:
+      "This is a reminder message from {ORGANIZER} of {EVENT_NAME} to {ATTENDEE} starting here {LOCATION} {MEETING_URL} at {START_TIME_h:mma} {TIMEZONE}.",
+    required: false,
+  })
+  @IsOptional()
+  @IsString()
+  text?: string;
+}
+
+export class WorkflowStepDto {
+  @ApiProperty({ description: "Unique identifier of the step", example: 67244 })
+  @IsNumber()
+  id!: number;
```

Comment:
Curious - why does an `id` need to be provided by customer? Can't we generate it?